handler/src/test/java/io/netty/handler/ssl/OpenSslErrorStackAssertSSLEngine.java

```diff
@@ -0,0 +1,440 @@
+            assertErrorStackEmpty();
+        }
+    }
+
+    @Override
+    public SSLParameters getSSLParameters() {
+        try {
+            return getWrappedEngine().getSSLParameters();
+        } finally {
+            assertErrorStackEmpty();
+        }
+    }
+
+    @Override
+    public void setSSLParameters(SSLParameters params) {
+        try {
+            getWrappedEngine().setSSLParameters(params);
+        } finally {
+            assertErrorStackEmpty();
+        }
+    }
+
+    public String getApplicationProtocol() {
+        if (PlatformDependent.javaVersion() >= 9) {
+            try {
+                return Java9SslUtils.getApplicationProtocol(getWrappedEngine());
+            } finally {
+                assertErrorStackEmpty();
+            }
+        }
+        throw new UnsupportedOperationException();
+    }
+
+    public String getHandshakeApplicationProtocol() {
+        if (PlatformDependent.javaVersion() >= 9) {
+            try {
+                return Java9SslUtils.getHandshakeApplicationProtocol(getWrappedEngine());
+            } finally {
+                assertErrorStackEmpty();
+            }
+        }
+        throw new UnsupportedOperationException();
+    }
+
+    public void setHandshakeApplicationProtocolSelector(BiFunction<SSLEngine, List<String>, String> selector) {
+        if (PlatformDependent.javaVersion() >= 9) {
+            try {
+                Java9SslUtils.setHandshakeApplicationProtocolSelector(getWrappedEngine(), selector);
+            } finally {
+                assertErrorStackEmpty();
+            }
+        }
+        throw new UnsupportedOperationException();
+    }
+
+    public BiFunction<SSLEngine, List<String>, String> getHandshakeApplicationProtocolSelector() {
+        if (PlatformDependent.javaVersion() >= 9) {
+            try {
+                return Java9SslUtils.getHandshakeApplicationProtocolSelector(getWrappedEngine());
+            } finally {
+                assertErrorStackEmpty();
+            }
+        }
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int refCnt() {
+        return getWrappedEngine().refCnt();
+    }
+
+    @Override
+    public OpenSslErrorStackAssertSSLEngine retain() {
+        getWrappedEngine().retain();
+        return this;
+    }
+
+    @Override
+    public OpenSslErrorStackAssertSSLEngine retain(int increment) {
+        getWrappedEngine().retain(increment);
+        return this;
+    }
+
+    @Override
+    public OpenSslErrorStackAssertSSLEngine touch() {
+        getWrappedEngine().touch();
+        return this;
+    }
+
+    @Override
+    public OpenSslErrorStackAssertSSLEngine touch(Object hint) {
+        getWrappedEngine().touch(hint);
+        return this;
+    }
+
+    @Override
+    public boolean release() {
+        return getWrappedEngine().release();
+    }
+
+    @Override
+    public boolean release(int decrement) {
+        return getWrappedEngine().release();
```

Comment:
missing param?

Reply:
ups... will fix it quickly.

Reply:
https://github.com/netty/netty/commit/47985c11c1d9f11b94018539108ed65f0c6a32ab